pkg/volume/csi/csi_plugin.go

```diff
@@ -124,6 +128,27 @@ func (p *csiPlugin) NewMounter(
 	return mounter, nil
 }
 
+func getReadOnlyOfVolumeMount(pod *api.Pod, spec *volume.Spec) (bool, error) {
+	if spec.PersistentVolume.Spec.ClaimRef == nil {
+		return false, errors.New("persistent volume has empty ClaimRef")
+	}
+	claimName := spec.PersistentVolume.Spec.ClaimRef
+	for _, v := range pod.Spec.Volumes {
+		if v.VolumeSource.PersistentVolumeClaim.ClaimName != claimName.Name {
+			continue
+		}
+		volName := v.Name
+		for _, c := range pod.Spec.Containers {
+			for _, vm := range c.VolumeMounts {
+				if vm.Name == volName {
+					return vm.ReadOnly, nil
```

Comment:
Technically `spec.ReadOnly` should be set to this value, but I believe there is a bug in the creation of `volume.Spec` in `volumemanager. DesiredStateOfWorldPopulator. createVolumeSpec(...)` <- the value should be set there but it is not.

I'm ok accepting this as a temporary fix and opening an issue to fix `spec.Readonly` later.

Reply:
Spoke with @sbezverk offline who explained that this is not true. `createVolumeSpec(...)` sets the `Readonly` using the readonly value from the `PVCVolumeSource`:

```GO
		// Fetch actual PV object
		volumeSpec, volumeGidValue, err :=
			dswp.getPVSpec(pvName, pvcSource.ReadOnly, pvcUID)
```

And in the case that a volume is referenced directly (without PVC/PV), it is set to the default (false, e.g. `readwrite`) because it is unclear what the correct value should be. In that case the last mile bind between mount point and container uses the pod container mount readonly flag which makes it appear correctly as readonly or not inside the container.